changelog/0075-reindex-all-indexes-for-es-upgrade.yml

```
@@ -0,0 +1,85 @@
+title: reindex-all-indexes-for-es-upgrade
+key: reindex-all-indexes-for-es-upgrade
+date: 2023-10-25
+optional_per_env: no
+# (optional) Min version of HQ that MUST be deployed before this change can be rolled out (commit hash)
+min_commcare_version:
+# I will add it after the PR is merged
+max_commcare_version:
+context: |
+  Reindex Elasticsearch Indices for upcoming ES Upgrade.
+details: |
+  Currently Commcare HQ is running with Elasticsearch 2.4. Reindexing is required before upgrading to a newer version of Elasticsearch 5.6.16
+update_steps: |
+  1. Deploy latest version of Commcare HQ.
+  2. Ensure that there is enough free space available in your cluster. The command to estimate disk space required for reindexing is:
+
+  ```sh
+  cchq <env> django-manage elastic_sync_multiplexed estimated_size_for_reindex
+  ```
+  3. Check disk usage on each node
+    - If you have sepearate data nodes, check disk usage on data nodes
+      ```sh
+      cchq <env>  run-shell-command es_data "df -h /opt/data" -b
+      ```
+    - If you don't have separate data nodes, check disk usage on ES nodes.
+      ```sh
+      cchq <env>  run-shell-command es_data "df -h /opt/data" -b
+      ```
+    This will return disk usage for each node. You can check if the cumulative available space across all nodes is greater than the total recommended space from the `estimated_size_for_reindex` output.
+
+  4. It is advised to run the following steps in a tmux session as they might take a long time and can be detached/re-attached as needed for monitoring progress.
+    - To start a tmux session in your django-manage machine, following steps can be followed, change `<env_name>` with your actual environment name:
+      ```bash
+      cchq <env> tmux django_manage
+      sudo -iu cchq
+      cd /home/cchq/www/<env>/current
+      source python_env/bin/activate
+      ```
+  
+  5. The following steps should be repeated for `${INDEX_CNAME}` serially set to following `['apps', 'cases', 'case_search', 'domains', 'forms', 'groups', 'sms', 'users']`
+
+    1. Set the `${INDEX_CNAME}` environment variable, for the first run it should be like
+    ```bash
+    INDEX_CNAME='apps'
+    ```
+  
+    2. Start the reindex process
+
+    ```bash
+    ./manage.py elastic_sync_multiplexed start ${INDEX_CNAME}
+    ```
+    Note down the Task Number that is displayed by the command. It should be a numeric ID and will be required to verify the reindex process.
+
+    3. Verify reindex is completed by querying the logs and ensuring that doc count matches between primary and secondary indices. The commands in this step should be run from control machine.
+
+      1. In a new shell login to control machine.
```

```suggestion
      1. Login to the control machine in a new shell.
```

Also, formatting of this sub-list does not seem to be rendering correctly on github. It's rendering as a fixed-width code block, which might mean it's indented too far?